pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarStreamMetadataProvider.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.stream.pulsar;
+
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.apache.pinot.spi.stream.OffsetCriteria;
+import org.apache.pinot.spi.stream.StreamConfig;
+import org.apache.pinot.spi.stream.StreamMetadataProvider;
+import org.apache.pinot.spi.stream.StreamPartitionMsgOffset;
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class PulsarStreamMetadataProvider extends PulsarPartitionLevelConnectionHandler implements StreamMetadataProvider {
+  private Logger LOGGER = LoggerFactory.getLogger(PulsarStreamMetadataProvider.class);
+
+  private String _clientId;
+  private StreamConfig _streamConfig;
+  private int _partition;
+
+  public PulsarStreamMetadataProvider(String clientId, StreamConfig streamConfig) {
+    super(clientId, streamConfig, 0);
+    _clientId = clientId;
+    _streamConfig = streamConfig;
+  }
+
+  public PulsarStreamMetadataProvider(String clientId, StreamConfig streamConfig, int partition) {
+    super(clientId, streamConfig, partition);
+    _clientId = clientId;
+    _streamConfig = streamConfig;
+    _partition = partition;
+  }
+
+  @Override
+  public int fetchPartitionCount(long timeoutMillis) {
+    try {
+      return _pulsarClient.getPartitionsForTopic(_streamConfig.getTopicName()).get().size();
+    } catch (Exception e) {
+      //TODO: Handle error
+      return 0;
+    }
+  }
+
+  public synchronized long fetchPartitionOffset(@Nonnull OffsetCriteria offsetCriteria, long timeoutMillis) {
+    throw new UnsupportedOperationException("The use of this method is not supported");
+  }
+
+  @Override
+  public StreamPartitionMsgOffset fetchStreamPartitionOffset(@Nonnull OffsetCriteria offsetCriteria,
```

Comment:
even this should not be implemented. Only `computePartitionGroupMetadata`

Reply:
done